service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/VerifyAzureStorageContainerCanBeCreatedStep.java

```diff
@@ -55,22 +70,32 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
     final StorageManager storageManager =
         crlService.getStorageManager(azureCloudContext, azureConfig);
 
+    UUID landingZoneId = null;
     try {
-      final WsmResource wsmResource =
-          resourceDao.getResource(resource.getWorkspaceId(), resource.getStorageAccountId());
-      final ControlledAzureStorageResource storageAccount =
-          wsmResource
-              .castToControlledResource()
-              .castByEnum(WsmResourceType.CONTROLLED_AZURE_STORAGE_ACCOUNT);
+      if (resource.getStorageAccountId() != null) {
```

Comment:
I think this code would be more readable if you separated the cases. Perhaps make one method for resolving the storage account name when `getStorageAccountId() != null` and a separate one when it is using the LZ. The mixing of error handling and fall-through is tough to parse.